cookbook/core/containerization/image_spec_example.py

```diff
@@ -0,0 +1,116 @@
+"""
+.. _image_spec_example:
+
+Building Docker Images without a Dockerfile
+-------------------------------------------
+
+.. tags:: Containerization, Intermediate
+
+Image Spec is a way to specify how to build a container image without a Dockerfile. The image spec by default will be
+converted to an `Envd <https://envd.tensorchord.ai/>`__ config, and the `Envd builder
+<https://github.com/flyteorg/flytekit/blob/master/plugins/flytekit-envd/flytekitplugins/envd
+/image_builder.py#L12-L34>`__ will build the image for you. However, you can also register your own builder to build
+the image using other tools.
+
+For every :py:class:`flytekit.PythonFunctionTask` task or a task decorated with the ``@task`` decorator,
+you can specify rules for binding container images. By default, flytekit binds a single container image, i.e.,
+the `default Docker image <https://ghcr.io/flyteorg/flytekit>`__, to all tasks. To modify this behavior,
+use the ``container_image`` parameter available in the :py:func:`flytekit.task` decorator, and pass an
+``ImageSpec``.
+
+Before building the image, Flytekit checks the container registry first to see if the image already exists. By doing
+so, it avoids having to rebuild the image over and over again. If the image does not exist, flytekit will build the
+image before registering the workflow, and replace the image name in the task template with the newly built image name.
+
+"""
+
+import pandas as pd
+from flytekit import ImageSpec, task, workflow, Resources
+# %%
+# .. admonition:: Prerequisites
+#    :class: important
+#
+#    - Install `flytekitplugins-envd <https://github.com/flyteorg/flytekit/tree/master/plugins/flytekit-envd>`__ to build the image spec.
+#    - To build the image on remote machine, check this `doc <https://envd.tensorchord.ai/teams/context.html#start-remote-buildkitd-on-builder-machine>`__
+#
+
+
+# %%
+# You can specify python packages, apt packages, and environment variables in the ``ImageSpec``.
+# These specified packages will be added on top of the `default image <https://github.com/flyteorg/flytekit/blob/master/Dockerfile>`__, which can be found in the Flytekit Dockerfile.
+# If desired, you can also override the default image by providing a custom ``base_image`` parameter when using the ``ImageSpec``.
+pandas_image_spec = ImageSpec(
+    base_image="ghcr.io/flyteorg/flytekit:py3.8-1.6.0",
```

Comment:
should we mention `DefaultImages.default_image()` also?

Reply:
Added it.